doc: nrf: releases: migration: 3.4: Add SRAM migration notes (#28681)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_3.4.rst

Build and configuration system
==============================

> * The Kconfig options :kconfig:option:`CONFIG_SRAM_SIZE` and :kconfig:option:`CONFIG_SRAM_BASE_ADDRESS` have been deprecated, boards should instead use the devicetree ``zephyr.sram`` chosen node to specify the RAM node which will be used (whose values populated the Kconfig values).

Reviewer: ```suggestion
* The Kconfig options :kconfig:option:`CONFIG_SRAM_SIZE` and kconfig:option:`CONFIG_SRAM_BASE_ADDRESS` have been deprecated.
  Use the devicetree ``zephyr.sram`` chosen node to specify which RAM node is used.
```

Author: fixed

>   If either option is manually adjusted, it will cause :kconfig:option:`CONFIG_SRAM_DEPRECATED_KCONFIG_SET` to be set which indicates this deprecation, however applications will continue to build and work with this notice.

Reviewer: ```suggestion
  If you adjust either option manually, the :kconfig:option:`CONFIG_SRAM_DEPRECATED_KCONFIG_SET` is set to indicate the deprecation.
  However, applications will continue to build and work with this notice.
```

Author: fixed

>   For the vast majority of cases, these values should not be changed as they default to the values of the ``zephyr,sram`` chosen node.

Reviewer: ```suggestion
  For the majority of cases, you should not change these values as they default to the values of the ``zephyr,sram`` chosen node.
```

Author: fixed

>   If code references these Kconfigs then they should be updated, no deprecation warning will be emitted when these values are referenced due to how Kconfig define generation works.

Reviewer: ```suggestion
  If the code references these Kconfig options, you should update them.
  No deprecation warning will be emitted when these values are referenced due to the Kconfig define generation process.
```

Author: fixed

  .. note::

>       This is listed in recommended changes for |NCS| v3.4 but will be a required change with |NCS| v3.5.

Reviewer: ```suggestion
  .. note::
     This is listed in the recommended changes for the |NCS| v3.4.
     In the |NCS| v3.5, this will be a required change.
```

Author: fixed

Build and configuration system
==============================

> * The Kconfig options :kconfig:option:`CONFIG_SRAM_SIZE` and kconfig:option:`CONFIG_SRAM_BASE_ADDRESS` have been deprecated.

Reviewer: ```suggestion
* The Kconfig options :kconfig:option:`CONFIG_SRAM_SIZE` and :kconfig:option:`CONFIG_SRAM_BASE_ADDRESS` have been deprecated.
```

  Use the devicetree ``zephyr.sram`` chosen node to specify which RAM node is used.
  If you adjust either option manually, :kconfig:option:`CONFIG_SRAM_DEPRECATED_KCONFIG_SET` is set to indicate the deprecation.
  However, applications will continue to build and work with this notice.
  For the majority of cases, you should not change these values as they default to the values of the ``zephyr,sram`` chosen node.
  If the code references these Kconfig options, you should update them.
  No deprecation warning will be emitted when these values are referenced due to the Kconfig define generation process.

  .. note::
     This is listed in the recommended changes for the |NCS| v3.4.

>      In the |NCS| v3.5, this will be a required change.

Reviewer 2: ```suggestion
     This is listed in the recommended changes for this |NCS| release.
     In the next release, this will be a required change.
```
As we normally don't mention the release version before those are officially released.

Author: Updated
